machine_learning_hep/data/data_prod_20200304/database_systematics_D0pp.yml

```diff
@@ -0,0 +1,134 @@
+categories:
+    default:
+        activate: true
+        label: "default"
+        variations:
+            default:
+                activate: true
+                label: "default"
+                diffs:
```

Comment:
Not sure at the moment, but does this work? Or do you have to put `diffs: {}`, meaning an empty dictionary? Or is this the default value anyway? Or is the default value `None`?
Sorry, really don't know at the moment. Might add a few more comments later today

Reply:
Thanks @benedikt-voelkel for pointing this out. According to the [documentation](https://yaml.org/spec/1.2/spec.html#id2786563) Empty nodes are resolved to a `null` value.
I [proceed to modify](https://github.com/ginnocen/MachineLearningHEP/pull/599/commits/a6a43c87098c222ba87f8c275f5d52f0643ce21c#diff-eb5b6943a57c9d3ffd166f38b4abb5b1R150) the database only if `diffs` is a dictionary, otherwise a warning is displayed and an unchanged database is saved.
In case `diffs` is an empty dictionary, the analysis parameters are not changed but the results paths are changed according to the category/variation names.
Maybe I could make it such that the paths are modified in both cases so that running a variation with empty diffs does not attempt to overwrite the default results (even though the analysis would stop anyway in such a case and prompt to delete the directories).
Let me know what you think is a more transparent behaviour.

Reply:
Another option might be to enforce that `diffs` is always a dictionary and abort processing of the variation database in case it isn't.